velox/functions/lib/string/StringImpl.h

```diff
@@ -673,4 +685,76 @@ FOLLY_ALWAYS_INLINE void pad(
       padPrefixByteLength);
 }
 
+// Converts the first character of each word in the input string to uppercase
+// The implementation logic is taken from
+// https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFInitCap.java.
+template <bool isAscii, typename TOutString, typename TInString>
+FOLLY_ALWAYS_INLINE bool initcap(
+    TOutString& outputStr,
+    const TInString& inputStr) {
+  if constexpr (isAscii) {
+    outputStr.resize(inputStr.size());
+    const char* inputChars = inputStr.data();
+    char* outputChars = outputStr.data();
+
+    bool isStartOfWord = true;
+
+    for (size_t index = 0; index < inputStr.size(); ++index) {
+      char currentChar = inputChars[index];
+
+      if (std::isspace(static_cast<unsigned char>(currentChar))) {
```

Comment:
Have a variable for static_cast<unsigned char>(currentChar) as that is used multiple times in this code.